rust/main/chains/hyperlane-ethereum/src/rpc_clients/fallback.rs

```diff
@@ -92,35 +93,102 @@ where
         R: DeserializeOwned,
     {
         use CategorizedResponse::*;
+
+        info!(method, "EthereumFallBackProvider method");
+
         let params = serde_json::to_value(params).expect("valid");
 
-        let mut errors = vec![];
-        // make sure we do at least 4 total retries.
-        while errors.len() <= 3 {
-            if !errors.is_empty() {
+        if method == "eth_sendRawTransaction" {
+            let mut i = 0; // retry if all providers returned a retryable error
+            let mut provider_errors;
+            loop {
+                let mut unordered = FuturesUnordered::new();
+                self.inner
+                    .providers
+                    .iter()
+                    .for_each(|p| unordered.push(Self::provider_request(p, method, &params)));
+
+                let mut errors = vec![];
+
+                loop {
+                    let Some(resp) = unordered.next().await else {
```

Comment:
nit: this can be a `while let Some(resp) = ...` instead of a loop

Reply:
Fixed